presto-parquet/src/main/java/io/prestosql/parquet/reader/ParquetReader.java

```diff
@@ -105,6 +109,13 @@ public ParquetReader(
         this.currentRowGroupMemoryContext = systemMemoryContext.newAggregatedMemoryContext();
         columns = messageColumnIO.getLeaves();
         this.options = requireNonNull(options, "options is null");
+        prestoTypes = new Type[columns.size()];
+        internalFields.stream()
+                .flatMap(ParquetReader::primitiveFields)
+                .forEach(field -> {
+                    checkArgument(prestoTypes[field.getId()] == null, "Field %s already assigned a type", field.getId());
+                    prestoTypes[field.getId()] = field.getType();
```

Comment:
is there always a corresponding presto type for internal parquet field? What if we read just subset of fields from parquet?

Reply:
Good catch, the assumption was not correct (not exactly here, but nonetheless)

Reply:
I added `Add column pruning test` commit & fixed the regression. Thanks!

Reply:
> So where was the code fixes for this?

https://github.com/prestosql/presto/pull/2840/files#diff-b0d2a903b8cfe527098dce1d1fee9badR303